### DISCUSSION TITLE
Compare the CheckSum between the GitHub release and the current Chocolatey package

### DISCUSSION THREAD

**Discussion by strausmann** [chocolatey-community/chocolatey-au#64] [Q&A]
I have a problem with a Chocolatey package where the software maintainer in GutHub likes to simply change the assets for a GitHub release.

I have been using Get-GitHubRelease to get the GitHub release information for the latest release.

Now, unfortunately, one of the assets has simply been changed without publishing a new GitHub release.

I would now like to compare the source in GutHub, the current release, with the CheckSum of the current Chocolatey package when running the update.ps1 script, even if the version number has not changed, i.e. no new GitHub release has been published.

What is the best way to do this? Thank you very much for your help?

**Comment by TheCakeIsNaOH** [comment]
One potential option would be to reach out to the software maintainer and ask that they follow best practices for releases by not changing the files, but instead bumping the version anytime there is an update. 

Other Potential options:
- Does the filename change? If so, you could compare the filename to the existing filename.
- The `updated_at` field from the releases url might be able to be recorded and then compared later, if the same filename is used. 
- If etags are provided for the files (I'm not sure for Github), that could be used to trigger a a checksum comparision. See spotify as an example: https://github.com/chocolatey-community/chocolatey-packages/blob/master/automatic/spotify/update.ps1

**Reply by strausmann** [reply]
Thank you very much for your feedback. Option 2 would also be possible. The software maintainer has at least used a new file name. I will also have a look at the options in the Spotify update.ps1.
